src/CounterInterceptor/Program.cs

```diff
@@ -0,0 +1,108 @@
+﻿namespace TemporalioSamples.CounterInterceptor;
+
+using Temporalio.Client;
+using Temporalio.Worker;
+
+internal class Program
+{
+    private static async Task Main(string[] args)
+    {
+        static string GetEnvVarWithDefault(string envName, string defaultValue)
+        {
+            string? value = Environment.GetEnvironmentVariable(envName);
+            if (string.IsNullOrEmpty(value))
+            {
+                return defaultValue;
+            }
+            return value;
+        }
+
+        var address = GetEnvVarWithDefault("TEMPORAL_ADDRESS", "127.0.0.1:7233");
+        var temporalNamespace = GetEnvVarWithDefault("TEMPORAL_NAMESPACE", "default");
+        var tlsCertPath = GetEnvVarWithDefault("TEMPORAL_TLS_CERT", string.Empty);
+        var tlsKeyPath = GetEnvVarWithDefault("TEMPORAL_TLS_KEY", string.Empty);
+        TlsOptions? tls = null;
+        if (!string.IsNullOrEmpty(tlsCertPath) && !string.IsNullOrEmpty(tlsKeyPath))

```

Comment:
I'm a little concerned about edge cases. If tlsCert/Key are not set, should the logic continue or should it be an early fail path. Since this is "main" and (I assume) running as an app, why not leave an error message and exit with an abnormal status.

I'd rather see `tls` set in a happy path after early conditions/assertions are all met for successfully moving forward.

Apologies if I'm off-base.

Reply:
Removed based on Chad's previous comment.